k8s/upgrade-job/src/upgrade/data_plane.rs

```diff
@@ -211,9 +265,29 @@ async fn is_data_plane_pod_running(
                 .build(),
             )?
             .as_str();
+
+        // skip if the the node name is not the same as the one on which pod needs to be restarted.
         if node != node_name {
             continue;
         } else {
+            // uid of io engine pod on the same node.
+            let io_engine_new_pod_uid = pod.metadata.uid.as_ref().ok_or(
+                EmptyPodUid {
+                    name: pod.name_any(),
+                    namespace: namespace.clone(),
+                }
+                .build(),
+            )?;
+
+            // valdiating that the pod uid on the node is same or not
+            // if same, that means that pod has not been started yet
+            // and new data plane pod is not up and running.
+            if io_engine_old_pod_uid == io_engine_new_pod_uid {
+                let pod_name = pod.name_any();
+                tracing::info!(node.name = %node,  pod.name = %pod_name , "Pod has not been deleted yet.");
```

Comment:
```suggestion
                tracing::debug!(node.name = %node,  pod.name = %pod_name , "Pod has not been deleted yet.");
```
```suggestion
                tracing::warn!(node.name = %node,  pod.name = %pod_name , "Pod has not been deleted yet.");
```
nit: Because this is a log message which does not provide much insight to a user, it is valuable only to the dev/QA.
In case, after serveral retries of checking  the Pod is not gone, we can conclusively log an error, else log a success message. Retry logging seems debug-ish.